src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -144,6 +144,17 @@ protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFact
             return null;
         }
 
+        public override bool InterestingForDynamicDependencyAnalysis
+        {
+            get
+            {
+                if(_type.IsInterface || _type.IsValueType || _type.Category == TypeFlags.Class)
+                    return _type.HasGenericVirtualMethod();
```

Comment:
Is this check really necessary? I guessed that we probably use the EETypeNode for array EETypes too, that's why I added the check for interface/class/valuetype